### PR TITLE
UW-477

### PR DESCRIPTION
**Pull request by maddenp-noaa**
**Synopsis**

Address [UW-477](https://jira.epic.oarcloud.noaa.gov/browse/UW-477):

- Add support for `attrs` key under `metatask`, supporting `mode` and `throttle` attributes.
- Update test and documentation.
- Update the `make format` command to format `.jsonschema` files without an intermediate temporary file, and in a way that halts a `make format && make test` sequence if `jq` encounters an error.

Preview documentation is [here](https://uwtools--386.org.readthedocs.build/en/386/).

**Type**

- [x] Documentation
- [x] Enhancement (adds a new functionality)

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: docs/sections/user_guide/uw_yaml/rocoto_yaml.rst [hunk 1]
```diff
@@ -621,3 +624,5 @@ The XML will look like this
 
      </metatask>
    </metatask>
+
+* The optional attributes ``mode`` and ``throttle`` are accepted under an ``attrs:`` key. See the :rocoto:`Rocoto documentation<>` for details.
```
Comment (NaureenBharwaniNOAA) [inline docs/sections/user_guide/uw_yaml/rocoto_yaml.rst:628]: Just wanted to double check that nothing is suppose to go between `Rocoto documentation<>`.  Possible I'm just not familiar with this syntax.

**Review by NaureenBharwaniNOAA — APPROVED** [review]
Everything looks great to me!

File: src/uwtools/resources/rocoto.jsonschema [hunk 2]
```diff
@@ -219,7 +219,7 @@
     },
     "metatask": {
       "additionalProperties": false,
-      "maxProperties": 3,
+      "maxProperties": 4,
```
Comment (WeirAE) [inline src/uwtools/resources/rocoto.jsonschema:222]: Boy I sure hope we don't find one with yet another layer...

File: Makefile [hunk 3]
```diff
@@ -32,7 +32,7 @@ format:
 	black src
 	isort src
 	cd src && docformatter . || test $$? -eq 3
-	for x in $$(find src -type f -name "*.jsonschema"); do jq -S . $$x >$$x.tmp && mv $$x.tmp $$x || rm $$x.tmp; done
+	for a in $$(find src -type f -name "*.jsonschema"); do b=$$(jq -S . $$a) && echo "$$b" >$$a || exit 1; done
```
[the remark this answers is not in the document]
Comment (WeirAE) [reply Makefile:35]: Ohh thats a good fix, something to definitely keep in mind

**Review by WeirAE — APPROVED** [review]
LGTM!

**Pull request merged**